src/controllers/hid/hidioreport.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+
+#include "controllers/controller.h"
+#include "controllers/hid/hiddevice.h"
+#include "util/duration.h"
+
+class HidIoReport {
+  public:
+    HidIoReport(const unsigned char& reportId,
+            hid_device* device,
+            std::shared_ptr<const mixxx::hid::DeviceInfo> deviceInfo);
+    void sendOutputReport(QByteArray data);
+
+  private:
+    const unsigned char m_reportId;
+    const RuntimeLoggingCategory m_logOutput;
+    hid_device* const
+            m_pHidDevice; // const pointer to the C data structure, which hidapi uses for communication between functions
+    std::shared_ptr<const mixxx::hid::DeviceInfo> m_pDeviceInfo;
```

Comment:
m_pDeviceInfo does nothing for the actual purpose of HidIOReport and is only used as context for logging?

Reply:
Correct!

Reply:
This is a known anti-pattern in software design.

Reply:
The domain classes should not be mixed up with unrelated infrastructure code. This needs to be decoupled.